src/main/java/com/icthh/xm/ms/entity/repository/backend/FileStorageRepository.java

```diff
@@ -0,0 +1,152 @@
+package com.icthh.xm.ms.entity.repository.backend;
+
+import com.google.common.hash.Hashing;
+import com.icthh.xm.commons.exceptions.BusinessException;
+import com.icthh.xm.commons.tenant.TenantContextHolder;
+import com.icthh.xm.ms.entity.config.ApplicationProperties;
+import com.icthh.xm.ms.entity.domain.Content;
+import com.icthh.xm.ms.entity.service.dto.UploadResultDto;
+import com.icthh.xm.ms.entity.util.XmHttpEntityUtils;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.*;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.icthh.xm.ms.entity.config.Constants.FILE_PREFIX;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+public class FileStorageRepository implements StorageRepository {
+
+    private final ApplicationProperties applicationProperties;
+    private final FilePrefixNameFactory  filePrefixNameFactory;
+    private final TenantContextHolder tenantContextHolder;
+
+    /**
+     * we should allow only letters, numbers and '/' symbol
+     */
+    private static final String FOLDER_NAME = "^[a-zA-Z0-9/]+$";
+    private static final Pattern pattern = Pattern.compile(FOLDER_NAME);
+
+    @Override
+    public String store(MultipartFile file, Integer size) {
+        return "";
+    }
+
+    @Override
+    @SneakyThrows
+    public String store(HttpEntity<Resource> httpEntity, Integer size) {
+        // Get the resource from HttpEntity
+        Resource resource = httpEntity.getBody();
+
+        if (resource == null) {
+            log.error("Resource in HttpEntity is null");
+            throw new BusinessException("storage.file.validation", "Resource in HttpEntity is null");
+        }
+
+        // Get the file template path from configurationC
+        String randomFilePrefix = Hashing.murmur3_128().hashString(UUID.randomUUID().toString(), StandardCharsets.UTF_8).toString();
```

Comment:
copy paste, pls extract method

Reply:
ok.
but we trade 2 lines on 3 lines, + 1 function call

Reply:
fixed